File: HackerRank/Algorithm/DiscreteMath/connecting_tower.pl
#!/usr/bin/perl
use strict;
use warnings;

sub routes
{
	my ($tower,$routes) = @_;

	my $output = 1;
	map { $output = $output*$_} split(' ',$routes);
	return ($output%1234567)
}

#Tail starts here
my $test_cases = <>;
for (my $index = 0; $index < $test_cases; $index++)
{
    my $keyword = <>;
    chomp $keyword;
    my $input = <>;
    chomp $input;
	print routes($keyword,$input);
	print "\n";
}

File: HackerRank/Algorithm/DiscreteMath/connecting_tower.py
#!/usr/bin/env python3

def find_ways():
    test_cases 	= int(input())
    for i in range(test_cases):
        num_integers = int(input())
        str_in 	= str(input()).strip()
        int_list = [int(x) for x in str_in.split(' ')]
        output = 1 
        for j in (int_list):
            output = output * j
        print (output%1234567)

def main():
    find_ways()
    
if __name__ == '__main__':
    main()